Response entity too large exception thrown for documents size more than 1 MB

CosmosDB [supports documents upto 2 MB][1]. However when create a document of size 1.5MB results in following error to be thrown

```
Caused by: io.netty.handler.codec.TooLongFrameException: Response entity too large: HttpObjectAggregator$AggregatedFullHttpResponse(decodeResult: success, version: HTTP/1.1, content: CompositeByteBuf(ridx: 0, widx: 1048576, cap: 1048576, components=128))
HTTP/1.1 201 Created
Cache-Control: no-store, no-cache
Pragma: no-cache
Content-Type: application/json
Server: Microsoft-HTTPAPI/2.0
Strict-Transport-Security: max-age=31536000
x-ms-last-state-change-utc: Tue, 24 Apr 2018 04:40:08.198 GMT
etag: "2400d364-0000-0000-0000-5adec0d30000"
x-ms-resource-quota: documentSize=10240;documentsSize=10485760;documentsCount=-1;collectionSize=10485760;
x-ms-resource-usage: documentSize=1;documentsSize=1537;documentsCount=0;collectionSize=1537;
x-ms-schemaversion: 1.6
x-ms-alt-content-path: dbs/java.rx.com.microsoft.azure.cosmosdb.rx.DocumentCrudTest/colls/7dba480a-3a92-48a0-ac8d-5ea39e754959
x-ms-quorum-acked-lsn: 7
x-ms-current-write-quorum: 3
x-ms-current-replica-set-size: 4
x-ms-xp-role: 1
x-ms-global-Committed-lsn: 7
x-ms-number-of-read-regions: 0
x-ms-transport-request-id: 12
x-ms-share-throughput: true
x-ms-request-charge: 1252.67
x-ms-serviceversion: version=1.21.0.0
x-ms-activity-id: ac53339f-1ce8-4716-8749-eac4311b03de
x-ms-session-token: 4:8
x-ms-gatewayversion: version=1.21.0.0
Date: Tue, 24 Apr 2018 05:29:54 GMT
Connection: keep-alive
    at io.netty.handler.codec.http.HttpObjectAggregator.handleOversizedMessage(HttpObjectAggregator.java:283)
    at io.netty.handler.codec.http.HttpObjectAggregator.handleOversizedMessage(HttpObjectAggregator.java:87)
    at io.netty.handler.codec.MessageAggregator.invokeHandleOversizedMessage(MessageAggregator.java:383)
    at io.netty.handler.codec.MessageAggregator.decode(MessageAggregator.java:277)
    at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:88)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
    at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:438)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:297)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:413)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
    at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
    at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
    at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1389)
    at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1159)
    at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1203)
    at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
    at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
    at io.reactivex.netty.metrics.BytesInspector.channelRead(BytesInspector.java:59)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
    at io.reactivex.netty.pipeline.InternalReadTimeoutHandler.channelRead(InternalReadTimeoutHandler.java:108)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1414)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:945)
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:146)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
    at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    ... 1 more
```

This happens because by default Netty sets [the size to 1 MB][2] and sdk uses the default constructor. Would be better to set it to 2 MB

PR to follow

[1]: https://docs.microsoft.com/en-us/azure/cosmos-db/sql-api-resources#documents
[2]: https://github.com/ReactiveX/RxNetty/blob/v0.4.20/rxnetty/src/main/java/io/reactivex/netty/protocol/http/HttpObjectAggregationConfigurator.java#L36

--- a/sdk/src/main/java/com/microsoft/azure/cosmosdb/rx/internal/RxDocumentClientImpl.java
+++ b/sdk/src/main/java/com/microsoft/azure/cosmosdb/rx/internal/RxDocumentClientImpl.java
@@ -27,6 +27,7 @@ import static com.microsoft.azure.cosmosdb.BridgeInternal.toDatabaseAccount;
 import static com.microsoft.azure.cosmosdb.BridgeInternal.toFeedResponsePage;
 import static com.microsoft.azure.cosmosdb.BridgeInternal.toResourceResponse;
 import static com.microsoft.azure.cosmosdb.BridgeInternal.toStoredProcedureResponse;
+import static org.apache.commons.io.FileUtils.ONE_MB;
 
 import java.io.IOException;
 import java.io.InputStream;
@@ -271,7 +272,7 @@ public class RxDocumentClientImpl implements AsyncDocumentClient, IAuthorization
                         MAX_REQUEST_HEADER_SIZE,
                         HttpClientPipelineConfigurator.MAX_CHUNK_SIZE_DEFAULT,
                         true),
-                new HttpObjectAggregationConfigurator<>());
+                new HttpObjectAggregationConfigurator((int)(ONE_MB * 2)));
         return clientPipelineConfigurator;
     }
 

--- a/sdk/src/test/java/com/microsoft/azure/cosmosdb/rx/DocumentCrudTest.java
+++ b/sdk/src/test/java/com/microsoft/azure/cosmosdb/rx/DocumentCrudTest.java
@@ -26,6 +26,7 @@ import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
+import org.apache.commons.lang3.StringUtils;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Factory;
@@ -42,6 +43,8 @@ import com.microsoft.azure.cosmosdb.rx.AsyncDocumentClient;
 import com.microsoft.azure.cosmosdb.rx.AsyncDocumentClient.Builder;
 
 import rx.Observable;
+
+import static org.apache.commons.io.FileUtils.ONE_MB;
 
 public class DocumentCrudTest extends TestSuiteBase {
 
@@ -61,6 +64,24 @@ public class DocumentCrudTest extends TestSuiteBase {
     @Test(groups = { "simple" }, timeOut = TIMEOUT)
     public void createDocument() throws Exception {
         Document docDefinition = getDocumentDefinition();
+
+        Observable<ResourceResponse<Document>> createObservable = client
+                .createDocument(getCollectionLink(), docDefinition, null, false);
+
+        ResourceResponseValidator<Document> validator = new ResourceResponseValidator.Builder<Document>()
+                .withId(docDefinition.getId())
+                .build();
+
+        validateSuccess(createObservable, validator);
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void createLargeDocument() throws Exception {
+        Document docDefinition = getDocumentDefinition();
+
+        //Keep size as ~ 1.5MB to account for size of other props
+        int size = (int) (ONE_MB * 1.5);
+        docDefinition.set("largeString", StringUtils.repeat("x", size));
 
         Observable<ResourceResponse<Document>> createObservable = client
                 .createDocument(getCollectionLink(), docDefinition, null, false);

